### PR TITLE
fix(popup): placement issue for android

### PR DESCRIPTION
**Pull request by Xiaoy312**
GitHub Issue (If applicable): #4761

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

Popup doesnt open at the right location in android. It is offset'd by the height of status bar.

## What is the new behavior?

Should behave like UWP now.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.Foundation/Point.cs [hunk 1]
```diff
@@ -20,6 +20,16 @@ public Point(double x, double y)
 			Y = y;
 		}
 
+#if __ANDROID__
+		internal static Point From(Action<int[]> getter)
```
Comment (davidjohnoliver) [inline src/Uno.Foundation/Point.cs:24]: Please move this method to `PopupPanel`, it's too specific to make sense as a method on `Point`.

**Review by davidjohnoliver — COMMENTED** [review]
Missing an automated test

File: src/Uno.Foundation/Point.cs (hunk 1 above)
Comment (Xiaoy312) [reply src/Uno.Foundation/Point.cs:24]: I think this belongs in `Point`. There is multiple places in uno.ui where we pass a reference to `new int[2]` to `.GetLocation(OnScreen|InWindow)`. Having this helper method would simplify things a little:
```cs
var p = new int[2];
view.GetLocationInWindow(p);
```
->
```cs
var p = Point.From(view.GetLocationInWindow);
```

**Comment by Xiaoy312** [comment]
The relevant automated tests can be found in #4757. This PR just addresses the android side of issue found during testing.

File: src/Uno.UI/UI/Xaml/Controls/Popup/PopupPanel.cs [hunk 2]
```diff
@@ -106,6 +106,23 @@ protected override Size ArrangeOverride(Size finalSize)
...
+#if __ANDROID__
+				// for android, the above line returns the absolute coordinates of anchor on the screen
+				// because the parent view of this PopupPanel is a PopupWindow and GetLocationInWindow will be (0,0)
+				// therefore, we need to make the relative adjustment
+				if (this.GetParent() is Android.Views.View view)
+				{
+					var windowLocation = Point.From(view.GetLocationInWindow);
+					var screenLocation = Point.From(view.GetLocationOnScreen);
+
+					if (windowLocation == default)
```
Comment (carldebilly) [inline src/Uno.UI/UI/Xaml/Controls/Popup/PopupPanel.cs:119]: Should epsilons be used here for the comparison?

Comment (Xiaoy312) [reply src/Uno.UI/UI/Xaml/Controls/Popup/PopupPanel.cs:119]: There is no need here. Since we are comparing `Point`, and not `PointF`.

**Review by carldebilly — APPROVED** [review]

**Review by davidjohnoliver — APPROVED** [review]

**Pull request merged**